### PR TITLE
pkg: fix typos in comments and tests

### PR DESCRIPTION
**Pull request by alexandear**
This PR fixes spelling mistakes in comments and tests in `pkg` directory.

- [ ] Ambient
- [ ] Configuration Infrastructure
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [x] Test and Release
- [] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by hanxiaop — APPROVED** [review]

**Review by ymesika — APPROVED** [review]

File: pkg/workloadapi/workload.pb.go [hunk 1]
```diff
@@ -315,7 +315,7 @@ type Service struct {
 	// The target_port may be overridden on a per-workload basis.
 	Ports []*Port `protobuf:"bytes,5,rep,name=ports,proto3" json:"ports,omitempty"`
 	// Optional; if set, the SAN to verify for TLS connections.
-	// Typically, this is not set and per-workload identity is used to verify
+	// Typically, this is not set and per-workload identity is used to verfiy
```
Comment (lei-tang) [inline pkg/workloadapi/workload.pb.go]: The "verfiy" in the PR is a typo. The original word "verify" is correct.

File: pkg/workloadapi/workload.proto [hunk 2]
```diff
@@ -70,7 +70,7 @@ message Service {
   // The target_port may be overridden on a per-workload basis.
   repeated Port ports = 5;
   // Optional; if set, the SAN to verify for TLS connections.
-  // Typically, this is not set and per-workload identity is used to verify
+  // Typically, this is not set and per-workload identity is used to verfiy
```
Comment (lei-tang) [inline pkg/workloadapi/workload.proto]: The "verfiy" in the PR is a typo. The original word "verify" is correct.

**Review by lei-tang — CHANGES_REQUESTED** [review]

File: pkg/workloadapi/workload.pb.go (hunk 1 above)
Comment (lei-tang) [reply pkg/workloadapi/workload.pb.go]: Reverted in later commit.

File: pkg/workloadapi/workload.proto (hunk 2 above)
Comment (lei-tang) [reply pkg/workloadapi/workload.proto]: Reverted in later commit.

**Review by lei-tang — APPROVED** [review]

**Pull request merged**